libs/shared/data-access/work-api/src/lib/models/form-configuration/form-configuration.model.ts

```diff
@@ -60,23 +60,11 @@ export const removeEmptyProperties = (obj: ExtendedComponentSchema): any => {
       })
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       .reduce((acc: any, [key, value]) => {
-        if (FORM_CONFIGURATION_KEYS_TO_KEEP.includes(key)) {
-          acc[key] = value;
-
-          return acc;
-        }
-
-        if (FORM_CONFIGURATION_KEYS_WITH_EMPTY_VALUES_TO_KEEP.includes(key) && !acceptValueType(value)) {
-          acc[key] = value;
-
-          return acc;
-        }
-
-        if (FORM_CONFIGURATION_KEYS_TO_REMOVE.includes(key)) {
-          return acc;
-        }
-
-        if (acceptValueType(value)) {
+        if (
+          FORM_CONFIGURATION_KEYS_TO_KEEP.includes(key) ||
+          (FORM_CONFIGURATION_KEYS_WITH_EMPTY_VALUES_TO_KEEP.includes(key) && !acceptValueType(value)) ||
+          (!FORM_CONFIGURATION_KEYS_TO_REMOVE.includes(key) && acceptValueType(value))
+        ) {
```

Comment:
Does it complain if we do it like this?, I think this is a little easier to read.

```suggestion
       if (FORM_CONFIGURATION_KEYS_TO_REMOVE.includes(key)) {
          return acc;
        }
        
        if (
          FORM_CONFIGURATION_KEYS_TO_KEEP.includes(key) ||
          (FORM_CONFIGURATION_KEYS_WITH_EMPTY_VALUES_TO_KEEP.includes(key) && !acceptValueType(value)) ||
          (acceptValueType(value))
        ) {
```